expression/builtin_arithmetic.go

```diff
@@ -273,13 +273,16 @@ func (s *builtinArithmeticPlusRealSig) Clone() builtinFunc {
 }
 
 func (s *builtinArithmeticPlusRealSig) evalReal(row chunk.Row) (float64, bool, error) {
-	a, isNull, err := s.args[0].EvalReal(s.ctx, row)
-	if isNull || err != nil {
-		return 0, isNull, err
+	a, isLHSNull, err := s.args[0].EvalReal(s.ctx, row)
+	if err != nil {
```

Comment:
Why change this?

Reply:
All of the expressions which have two arguments in TiDB have this pattern, they have inconsistent behavior with MySQL.
```
create table t (a varchar(10));
insert into t values ('2007');
select * from t where NULL + pow(7020, a);
```
The above statement should report an error.